File: 3.py
arr_name = [
    {"name": "Lisa"},
    {"name": "Bart"},
    {"name": "Maggie"},
    {"name": "Masha"}
]


def interestingMethod(arr=[]):
    names=[]
    finalString=''
    for item in arr:
      names.append(item.get('name'))
    endName=names.pop()
    for name in names:
      finalString+=name+', '
    return endName if len(names)==0 else finalString.removesuffix(', ')+'&'+endName


print(interestingMethod(arr_name))

File: 3.js
arrName = [
  { name: 'Lisa' },
  { name: 'Bart' },
  { name: 'Maggie' },
  { name: 'Masha' },
];

function interestingMethod(arr) {
  let tempString = arr.map((i) => i.name).join(', ');
  finalString =
    tempString.substring(0, tempString.lastIndexOf(', ')) +
    '&' +
    tempString.substring(tempString.lastIndexOf(', ') + 2);
  return finalString;
}

console.log(interestingMethod(arrName));
